bundles/binding/org.openhab.binding.powermax/src/main/java/org/openhab/binding/powermax/internal/PowerMaxBinding.java

```diff
@@ -0,0 +1,907 @@
+                updateItemsFromPanelSettings();
+                if (currentState.isPowerlinkMode()) {
+                    logger.info("PowerMax alarm binding: running in Powerlink mode");
+                    PowerMaxCommDriver.getTheCommDriver().sendMessage(PowerMaxSendType.RESTORE);
+                } else {
+                    logger.info("PowerMax alarm binding: running in Standard mode");
+                    PowerMaxCommDriver.getTheCommDriver().sendMessage(PowerMaxSendType.ZONESNAME);
+                    PowerMaxCommDriver.getTheCommDriver().sendMessage(PowerMaxSendType.ZONESTYPE);
+                    PowerMaxCommDriver.getTheCommDriver().sendMessage(PowerMaxSendType.STATUS);
+                }
+                PowerMaxCommDriver.getTheCommDriver().exitDownload();
+            }
+        }
+    }
+
+    /**
+     * Post item updates on the bus to match a new alarm system state
+     *
+     * @param selector
+     *            filter on selector type; if null, no filter on selector type
+     * @param state
+     *            the alarm system state
+     */
+    private void updateItemsFromAlarmState(PowerMaxSelectorType selector, PowerMaxState state) {
+        updateItemsFromAlarmState(null, null, selector, state);
+    }
+
+    /**
+     * Post item updates on the bus to match a new alarm system state
+     *
+     * @param state
+     *            the alarm system state
+     */
+    private void updateItemsFromAlarmState(PowerMaxState state) {
+        updateItemsFromAlarmState(null, null, null, state);
+    }
+
+    /**
+     * Post item updates on the bus to match a new alarm system state
+     *
+     * @param provider
+     *            filter on provider; if null, no filter on provider
+     * @param name
+     *            filter on item name; if null, no filter on item name
+     * @param selector
+     *            filter on selector type; if null, no filter on selector type
+     * @param state
+     *            the alarm system state
+     */
+    private synchronized void updateItemsFromAlarmState(PowerMaxBindingProvider provider, String name,
+            PowerMaxSelectorType selector, PowerMaxState state) {
+
+        if (state == null) {
+            return;
+        }
+        if (itemUpdateDisabled) {
+            logger.debug("updateItemsFromAlarmState(): items update disabled");
+            return;
+        }
+
+        if (provider == null) {
+            for (PowerMaxBindingProvider prov : providers) {
+                updateItemsFromAlarmState(prov, name, selector, state);
+            }
+        } else {
+            for (String itemName : provider.getItemNames()) {
+                if ((name == null) || itemName.equals(name)) {
+                    State itemState = null;
+
+                    PowerMaxBindingConfig config = provider.getConfig(itemName);
+                    Integer num = config.getSelectorIntParam();
+                    Item item = config.getItem();
```

Comment:
> I need to know the item type in oder to send on the bus ON / OFF for a switch item and OPEN / CLOSE for a contact item. The state object to be used is not the same.

You don't need to know the item type, but you need to know the list of types that an item can receive as state and commands.  So keep the lists, and use your own loops over the types, or use the TypeParser if it's a String object.

Examples of other bindings that do something similar:

* o.o.b.myq.internal.MyqBinding.java#L213
* o.o.b.garadget.internal.AbstractDevice.java#L134
* o.o.b.http.internal.HttpGenericBindingProvider.java#L299
* o.o.b.mqtt.internal.MqttMessageSubscriber.java#L65